avatars/avatars.js

```diff
@@ -1989,40 +2009,68 @@ class Avatar {
     // setup
     if (enabled) {
       this.volume = 0;
-     
+
       const audioContext = getAudioContext();
       if (audioContext.state === 'suspended') {
         (async () => {
           await audioContext.resume();
         })();
       }
-      this.microphoneWorker = new MicrophoneWorker({
-        audioContext,
-        muted: false,
-        emitVolume: true,
-        emitBuffer: true,
-      });
-      this.microphoneWorker.addEventListener('volume', e => {
-        if(!this.manuallySetMouth){
-          this.volume = this.volume*0.8 + e.data*0.2;
+      const _volume = e => {
+        if (!this.manuallySetMouth) {
+          this.volume = this.volume * 0.8 + e.data * 0.2;
+        } else {
+          console.log("couldn't set mouth because it is being manually set")
         }
-      });
-      this.microphoneWorker.addEventListener('buffer', e => {
+      }
+
+      const _buffer = e => {
         this.audioRecognizer.send(e.data);
-      });
+      }
 
       this.audioRecognizer = new AudioRecognizer({
         sampleRate: audioContext.sampleRate,
       });
+
       this.audioRecognizer.addEventListener('result', e => {
         this.vowels.set(e.data);
       });
+
+      this.audioWorker = new MicrophoneWorker({
```

Comment:
Why is there a second `MicrophoneWorker`?

Reply:
It's to handle the local mouth flap case which should not have audio feedback.

Reply:
Probably a better name for these workers would be `localAudioWorker` and `remoteAudioWorker`.